Index: varstored-1.0.0/README.md
===================================================================
--- varstored-1.0.0/README.md
+++ varstored-1.0.0/README.md	2022-10-25 15:02:10.861399195 +0200
@@ -46,8 +46,8 @@
 And place them where varstored can find them:
 
 ```
-$ mkdir -p /usr/share/varstored
-$ cp PK.auth KEK.auth db.auth dbx.auth /usr/share/varstored
+$ mkdir -p /var/lib/varstored
+$ cp PK.auth KEK.auth db.auth dbx.auth /var/lib/varstored
 ```
 
 Build the main daemon and tools and install them:
Index: varstored-1.0.0/handler.c
===================================================================
--- varstored-1.0.0/handler.c
+++ varstored-1.0.0/handler.c	2022-10-25 15:01:14.783119775 +0200
@@ -130,7 +130,7 @@
 static const uint8_t EFI_IMAGE_SECURITY_DATABASE1[] = {'d',0,'b',0,'x',0};
 static const uint8_t EFI_IMAGE_SECURITY_DATABASE2[] = {'d',0,'b',0,'t',0};
 
-#define AUTH_PATH_PREFIX "/usr/share/varstored"
+#define AUTH_PATH_PREFIX "/var/lib/varstored"
 
 /*
  * Array of auth_info structs containing the information about the keys
