diff --git a/handler.c b/handler.c
index 28a494f..1c1a539 100644
--- a/handler.c
+++ b/handler.c
@@ -2171,11 +2171,18 @@ setup_keys(void)
         if (!auth_info[i].data) {
             WARN("Cannot setup %s because auth data is missing!\n",
                  auth_info[i].pretty_name);
+
+            if (auth_info[i].name == EFI_IMAGE_SECURITY_DATABASE1) {
+                INFO("Continuing keys setup despite missing optional dbx\n");
+                continue;
+            }
+
             /*
              * Skip setting the rest of the keys (in particular, PK).
              * Otherwise the platform may be in user mode without
              * KEK/db set which will cause in-guest dbx updates to fail.
              */
+            WARN("Aborting keys setup\n");
             return true;
         }
 
